docs/hooks-script.md

```diff
@@ -0,0 +1,231 @@
+For instance, you can have both `OnBeforeConfigLoad` and `OnAfterConfigLoad`,
+to change the behaviour before and after the config are loaded.
+
+## Advanced hooks
+
+### How to change the behaviour of a function (filtering)
+
+With hooks you can't only run specific actions at a specific time, you can even change the behaviour of the script where the hook is called
+To do so, you have 2 solutions:
+
+#### 1) Using reference parameters
+
+This is the most common one. Basically using the concept of passing a parameter by reference you can change everything is passed to the hook itself.
+For instance:
+
+```C++
+OnMotdChange(std::string& newMotd)
+``` 
+
+Passing the newMotd with the '&' character you allow the listeners to change the value of the Motd when that action is called.
+
+#### 2) Using a bool return value
+
+This approach is not very common, most of the hooks return a "void" type and working with references is easier most of the time, but if you really need it you can implement a hook which is declared in this way:
+
+```C++
+bool ScriptMgr::OnBeforePlayerTeleport(Player* player, uint32 mapid, float x, float y, float z, float orientation, uint32 options, Unit* target)
+{
+    bool ret = true;
+    FOR_SCRIPTS_RET(PlayerScript, itr, end, ret) // return true by default if not scripts
+    if (!itr->second->OnBeforeTeleport(player, mapid, x, y, z, orientation, options, target))
+        ret = false; // we change ret value only when scripts return false
+
+    return ret;
+}
+```
+
+This hook notifies all the listeners but also catches when at least one of the registered listener returns "false", in that case the final return value will be false as well.
+In this particular case this hook is used within an if-condition to disallow a player to be teleported if one of the listener returns **false** for some reason.
+
+You can implement your different logic (e.g. false by default, true if any) just remember to document it properly!
+
+### Create your hook system within your module
+
+By using the guide above you can even create your ScriptObject within your module to allow people extending it. 
+Some modules, such as the autobalance, allows to customize certain part of their function by using internal hooks
+
+You can take a look at this file as example: https://github.com/azerothcore/mod-autobalance/blob/master/src/AutoBalance.h
+
+NOTE: You also need to create your own ScriptMgr implementation and offer a singleton to allow calling your hooks.
+
+
+### Final considerations
+
+There are different other features of the ScriptAI system that have not been included in this documentation, such as the creation of scripts binded 
+to specific entities inside our database (E.g. CreatureScript). This advanced usage can be implemented by replicate the related code we have inside the ScriptMgr files.
+If you need any help or you want to improve this documentation, feel free to ask support and edit this page.
+
```

Comment:
last note, mabe you can add some practical examples such as the one I added here:

https://stackoverflow.com/questions/66340549/is-it-possible-to-turn-a-core-patch-into-a-module-for-azerothcore/66340683#66340683

Reply:
I'm not sure what part of that post you link can be helpful for the reader of this wiki page and how to expose it. I could put the link directly